packages/synapse-react-client/src/components/EntityUpload/FileUploadProgress.tsx

```diff
@@ -0,0 +1,182 @@
+import {
+  CheckCircleTwoTone,
+  DeleteTwoTone,
+  ErrorTwoTone,
+  Pause,
+  PlayArrow,
+} from '@mui/icons-material'
+import CloseIcon from '@mui/icons-material/Close'
+import {
+  Box,
+  IconButton,
+  LinearProgress,
+  Tooltip,
+  Typography,
+} from '@mui/material'
+import React from 'react'
+import { calculateFriendlyFileSize } from '../../utils/functions/calculateFriendlyFileSize'
+
+export type FileUploadProgressProps = {
+  /** The status of the upload. */
+  status:
+    | 'PREPARING'
+    | 'UPLOADING'
+    | 'PAUSED'
+    | 'CANCELED_BY_USER'
+    | 'FAILED'
+    | 'COMPLETE'
+  /** The name of the file */
+  fileName: string
+  /** The size of the file, in bytes */
+  totalSizeInBytes: number
+  /** The number of this file's bytes uploaded so far. */
+  uploadedSizeInBytes?: number
+  /** Invoked when the upload is cancelled. */
+  onCancel: () => void
+  /** Invoked when the upload is paused. */
+  onPause: () => void
+  /** Invoked when the resumed. */
+  onResume: () => void
+  /** Invoked when the upload is removed from the list. */
+  onRemove: () => void
+  /** An optional error message to display if the upload has been cancelled due to error. */
+  errorMessage?: string
+}
+
+export const FILE_UPLOAD_PROGRESS_COMPONENT_HEIGHT_PX = 92
+
+/**
+ * Component that displays the upload progress of a file, with controls to pause or cancel the upload.
+ */
+export function FileUploadProgress(props: FileUploadProgressProps) {
```

Comment:
[Screencast from 11-13-2024 11:40:54 AM.webm](https://github.com/user-attachments/assets/c2bbbc11-e87f-47a9-9b32-1cabb19f82d0)

Reply:
Trouble loading this video (or downloading from the [source url](https://private-user-images.githubusercontent.com/17580037/385854041-c2bbbc11-e87f-47a9-9b32-1cabb19f82d0.webm?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1MjQ0MDcsIm5iZiI6MTczMTUyNDEwNywicGF0aCI6Ii8xNzU4MDAzNy8zODU4NTQwNDEtYzJiYmJjMTEtZTg3Zi00N2E5LTliMzItMWNhYmIxOWY4MmQwLndlYm0_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExM1QxODU1MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01ZjhiYmJhYzU3NzhmOWMxYTE1ZTg1YWYxYTZhNjlkNjRkNWU2N2UzZDQxZWRlZDIxMjA3OTFmOWI1Y2I5YTNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.LQEbyj8ictgp8r3Yu3oGIoqJpGuH1IVDjXbbzkVjdmk))

Reply:
The video was an animated capture of the various states of the FileUploadProgress component:

![image](https://github.com/user-attachments/assets/0b4d9ee4-c7ed-4db8-a984-f75dc02b36e6)